src/js/components/DataTable/doc.js

```diff
@@ -185,6 +185,14 @@ export const doc = DataTable => {
       names and values which are the search text strings. This is typically
       employed so a back-end can be used to search through the data.`,
     ),
+    onSelect: PropTypes.func.description(
```

Comment:
Using `onSelect` without the `select` property makes any sense?

Reply:
Allowed using it uncontrolled in this way.